documentation/configuration/parameters/encoding.md

```diff
@@ -9,7 +9,40 @@ redirect_from: /documentation/configuration/encoding/
 # Encoding
 
 ## Description
-The encoding of Sql migrations
+The encoding of Sql migrations.
+
+The encodings that Flyway supports are:
+
+- `US-ASCII`
+- `ISO-8859-1`
+- `UTF-8`
+- `UTF-16BE`
+- `UTF-16LE`
+- `UTF-16`
+
+All your scripts must have the same file encoding.
+
+### Encoding mismatches
+
+It's possible that Flyway will successfully read a script even if it doesn't match this configuration option.
+
+For instance, an `ISO-8859-1` encoded script might work if Flyway's encoding is set to `UTF-8`. However it's inevitable that a failure will occur when an un-decodable character appears.
+
+Therefore you must ensure all your scripts have the same encoding.
+
```

Comment:
I'm not sure I want to go into such detail about why some encodings might work instead of others. I don't want to give the impression that it's valid to mix UTF-8 and ISO-8859-1 for instance - because what is the advantage of doing so? Really users ought to be consistent.